pkgs/compile-pint-project.nix

```diff
@@ -0,0 +1,48 @@
+{ pkgs, pint }:
+pkgs.writeShellScriptBin "compile-pint" ''
+  display_help() {
+    echo "Usage: $0 [OPTIONS] <dir>"
+    echo "Compiles a pint project or directory of projects."
+    echo
+    echo "Options:"
+    echo "  -w, --workspace Compile a directory of projects"
+    echo "  -h, --help      Display this help message and exit"
+  }
+
+  is_workspace_flag() {
+    local arg="$1"
+    [ "$arg" == "--workspace" ] || [ "$arg" == "-w" ]
+  }
+
+  for arg in "$@"
+  do
+      if [ "$arg" == "--help" ] || [ "$arg" == "-h" ]
+      then
+          display_help
+          exit 0
+      fi
+  done
+
+  if [ -z "$1" ] || ( is_workspace_flag "$1" && [ -z "$2" ] ) || ( ! is_workspace_flag "$1" && [ -n "$2" ] ); then
+    display_help
+    exit 1
+  fi
+
+  if is_workspace_flag "$1"; then
+    cd "$2" || exit 1
+    for dir in */; do
+      if [ -d "$dir" ]; then
+        echo "compiling $dir"
+        cd "$dir" || continue
+        
+        ${pint}/bin/pint build
```

Comment:
Should we check here that `$dir` is not `target/`? 
It's fine if we don't because there it is harmless to try and the output looks like:
```
compiling target/
Error: no `pint.toml` in the current or parent directories
```
and then goes on with the execution.
If there were contracts under `target/`, would they be compiled and still live under this directory along with the pint output?

Reply:
This script will only build contracts at the directory you point it at so you need to do `nix run .#pint-proj apps/token/pint`. I should probably document this in the read me. Maybe I can add another script that uses this script to build all the contracts in the repo.